smartsim/_core/mli/infrastructure/control/requestdispatcher.py

```diff
@@ -0,0 +1,525 @@
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# pylint: disable=import-error
+# pylint: disable-next=unused-import
+import dragon
+from dragon.managed_memory import MemoryPool
+from dragon.mpbridge.queues import DragonQueue
+
+# pylint: enable=import-error
+
+# isort: off
+# isort: on
+
+import multiprocessing as mp
+import time
+import typing as t
+import uuid
+from queue import Empty, Full, Queue
+from threading import RLock
+from types import TracebackType
+
+from smartsim._core.entrypoints.service import Service
+
+from .....error import SmartSimError
+from .....log import get_logger
+from ....utils.timings import PerfTimer
+from ...comm.channel.channel import CommChannelBase
+from ...comm.channel.dragonchannel import DragonCommChannel
+from ...infrastructure.environmentloader import EnvironmentConfigLoader
+from ...infrastructure.storage.featurestore import FeatureStore
+from ...infrastructure.worker.worker import (
+    InferenceBatch,
+    InferenceRequest,
+    MachineLearningWorkerBase,
+)
+from ...message_handler import MessageHandler
+from ...mli_schemas.model.model_capnp import Model
+from ...mli_schemas.response.response_capnp import ResponseBuilder
+from ...mli_schemas.tensor.tensor_capnp import TensorDescriptor
+
+if t.TYPE_CHECKING:
+    from smartsim._core.mli.mli_schemas.response.response_capnp import Status
+
+logger = get_logger("Request Dispatcher")
+
+
+def deserialize_message(
+    data_blob: bytes,
+    channel_type: t.Type[CommChannelBase],
+) -> InferenceRequest:
+    """Deserialize a message from a byte stream into an InferenceRequest
+    :param data_blob: The byte stream to deserialize"""
+    # todo: consider moving to XxxCore and only making
+    # workers implement the inputs and model conversion?
+
+    # alternatively, consider passing the capnproto models
+    # to this method instead of the data_blob...
+
+    # something is definitely wrong here... client shouldn't have to touch
+    # callback (or batch size)
+
+    request = MessageHandler.deserialize_request(data_blob)
```

Comment:
Looks like all the comments have made the method obsolete and we could just use last line in place of where it was called